src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/external/UserController.kt

```diff
@@ -155,14 +160,94 @@ class UserController(
     @Parameter(
       description = "The reason user made inactive.",
       required = true
-    ) @org.springframework.web.bind.annotation.RequestBody
+    ) @RequestBody
     deactivateReason: DeactivateReason
   ) = userService.disableUserByUserId(
     userId,
     deactivateReason
   )
+
+  @PostMapping("/externalusers/{userId}/email")
+  @PreAuthorize("hasAnyRole('ROLE_MAINTAIN_OAUTH_USERS', 'ROLE_AUTH_GROUP_MANAGER')")
+  @Operation(
+    summary = "Amend a user email address.",
+    description = "Amend a user email address."
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "204",
+        description = "OK."
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Bad request e.g. if validation failed or if the email changes are disallowed",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Unable to amend user, the user is not within one of your groups or you don't have ROLE_MAINTAIN_OAUTH_USERS or ROLE_AUTH_GROUP_MANAGER roles",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "User not found.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      )
+    ]
+  )
+
+  fun alterUserEmail(
+    @Parameter(description = "The ID of the user.", required = true) @PathVariable
+    userId: UUID,
+    @RequestBody amendUser: AmendUser,
+    @Parameter(hidden = true) request: HttpServletRequest,
+  ): String? {
+    val setPasswordUrl = createInitialPasswordUrl(request)
+    val resetLink = userService.amendUserEmailByUserId(
+      userId,
+      amendUser.email,
+      setPasswordUrl
+    )
+    return if (smokeTestEnabled) resetLink else null
+  }
+
+  private fun createInitialPasswordUrl(@Parameter(hidden = true) request: HttpServletRequest): String {
+    val requestURL = request.requestURL
+    return requestURL.toString().replaceFirst("/api/authuser/.*".toRegex(), "/initial-password?token=")
```

Comment:
I don't think is quite right as the url is no longer /api/authuser

Reply:
Corrected - now uses `api.base.url.oauth` environment variable